app/models/search_builder/synthetic_category_limit.rb

```diff
@@ -0,0 +1,37 @@
+class SearchBuilder
+  # A search builder module to restrict to a certain SyntheticCollection.
+  # Won't do _anything_ just by being in the chain -- sets class_attributes
+  # on the SearchBuilder that can be used to turn it 'on', either on the class
+  # or even on an instance.
+  #
+  #.    synthetic_category_param : set to a string, and this will be taken
+  #.      from app params to identify a collection to limit to.  If unknown,
+  #.      will do nothing. (Should limit to 0 rows instead?)
```

Comment:
If the alternative is "don't filter at all" then I think yes 0 hits makes more sense. I see your point about that won't be possible, though; I did notice the ArgumentError for that case. So I'm fine to merge as is.